feat: add amendment summary page

pdp2121 opened the pull request:
## High Level Overview of Change

<!--
Please include a summary/list of the changes.
If too broad, please consider splitting into multiple PRs.
-->

The page would include:

- Information about the amendment
- List of voted Yes and No validators (for amendments in voting)
- A graph showing distribution of votes broken down by UNL and non-UNL validators (for amendments in voting)

This PR would also include search by Amendment Id/Name

### Type of Change

<!--
Please check relevant options, delete irrelevant ones.
-->

- [x] New feature (non-breaking change which adds functionality)

## Before / After

<!--
If just refactoring / back-end changes, this can be just an in-English description of the change at a technical level.
If a UI change, screenshots should be included.
-->

### Amendment in voting (Desktop)

![localhost_3001_amendment_Clawback (1)](https://github.com/ripple/explorer/assets/71317875/81e6bf71-29fa-4556-8234-46dff0810579)



### Amendment in voting (Mobile)


![localhost_3001_amendment_Clawback(iPhone 12 Pro) (1)](https://github.com/ripple/explorer/assets/71317875/b0c21b47-3cbb-492a-88de-5a35b6e9faae)



### Amendment enabled
![Screenshot 2023-10-03 at 7 09 56 PM](https://github.com/ripple/explorer/assets/71317875/74f282ba-d181-4a0d-9f87-ef7f3c3cd772)

### Chart legend

<img width="1294" alt="Screenshot 2023-10-04 at 11 49 39 AM" src="https://github.com/ripple/explorer/assets/71317875/819d8a40-11a2-49f2-8784-e1d0568a965b">

### Footnote

<img width="1298" alt="Screenshot 2023-10-04 at 4 36 09 PM" src="https://github.com/ripple/explorer/assets/71317875/4437b640-90d0-486e-8251-3241bab171d2">

bugsbunnies commented:
@pdp2121 The `Yeas` and `Nays` colors are reversed, should be Yeas - Green, Nays - Magenta

pdp2121 commented:
> @pdp2121 The `Yeas` and `Nays` colors are reversed, should be Yeas - Green, Nays - Magenta. In addition to the color fix, let's have the `Yeas` bar on the left and `Nays` on the right

@bugsbunnies Nice catch! I just updated the color and screenshots

mvadari commented:
The "UNLs"/"non-UNLs" labels look like they're supposed to be labeling each of the bars. Can those titles be moved to the top of each graph?

bugsbunnies commented:
> The "UNL"/"non-UNL" labels look like they're supposed to be labeling each of the bars. Can those titles be moved to the top of each graph?

I think the problem is that they're at an angle, can we just make it so they center align with each bar group and not at an angle?

mvadari commented:
> > The "UNL"/"non-UNL" labels look like they're supposed to be labeling each of the bars. Can those titles be moved to the top of each graph?
> 
> I think the problem is that they're at an angle, can we just make it so they center align with each bar group and not at an angle?

Yes, agreed - I think it'd look better with a top label, but I'm not opposed to a bottom axis label.

bugsbunnies commented:
> > > The "UNL"/"non-UNL" labels look like they're supposed to be labeling each of the bars. Can those titles be moved to the top of each graph?
> > 
> > 
> > I think the problem is that they're at an angle, can we just make it so they center align with each bar group and not at an angle?
> 
> Yes, agreed - I think it'd look better with a top label, but I'm not opposed to a bottom axis label.

I suggested bottom label just to keep it consistent with the basics of a graph with y-axis label and x-axis label in their expected places. The tricky thing with a top label (if im imagining what you're saying correctly) is that if we give them a fixed position on the graph, they might not always be close to the data bars, and if we keep them close to the data, the label could move around depending on the bars height.

mvadari commented:
Shouldn't the table only show UNL - or have tabs to show UNL vs non-UNL?

pdp2121 commented:
I have fixed the axis accordingly and updated the screenshots. The Simple of this page shows both Yeas(All), Nays(All), Yeas(UNL), and Nays(UNL), while the amendments list page in https://github.com/ripple/explorer/pull/836 only count UNL for voters.

intelliot commented:
Agree that it's useful to show all validators (instead of just UNL) and it makes sense to show all. 

dUNL validators should be marked with the green check. 

optional - It would also be useful to have a footnote to indicate what the green check means: "this validator is on the vl.ripple.com UNL" (or vl.xrplf.org, whichever is accurate).

pdp2121 commented:
Makes sense! I have moved the UNL validators to the top and added a footnote for clarification. I think this is clear enough to explain those columns.

ckniffen commented on src/containers/Header/Search.tsx:
```diff
@@ -36,9 +38,24 @@ const determineHashType = async (id: string, rippledContext: XrplClient) => {
     await getTransaction(rippledContext, id)
     return 'transactions'
   } catch (e) {
+    const isAmendment = await isAmendmentType(id)
```
Maybe this should happen in parallel since there are 3 options.  This is even further complicated by #544.  Doing in parallel makes sense to allow for #831. I think this should be moved out into a new PR so these other issues can maybe be addressed there too.

ckniffen commented on src/containers/Amendment/amendment.scss:
```diff
@@ -0,0 +1,138 @@
+  .votes {
+    margin: 48px 0;
+
+    .barchart {
```
Consider putting the `max-width` on `.votes`

ckniffen commented on src/containers/Amendment/index.tsx:
```diff
@@ -0,0 +1,168 @@
+  const { data, error, isLoading } = useQuery<
+    AmendmentVote,
+    keyof typeof ERROR_MESSAGES | null
+  >(['fetchAmendmentData', identifier], async () => fetchAmendmentData(), {
```
Should `network` be a part of the key too?

ckniffen commented on src/containers/Amendment/index.tsx:
```diff
@@ -0,0 +1,168 @@
+      .then((response) => {
+        if (response.voting_status === 'voting') {
+          return {
+            voting_status: 'voting',
```
Make voting_status options a constant.

ckniffen commented on public/locales/en-US/translations.json:
```diff
@@ -488,6 +488,30 @@
+  "not_enabled": "Not Enabled",
+  "enabled": "Enabled",
+  "enabled_on": "enabled on",
+  "yeas_all": "yeas (all)",
```
these three could be a combination of the two values so you dont need all 4.
